### ISSUE TITLE
"Federation -> XML to SimpleSAMLphp metadata converter" broken with Symfony Theme

### ISSUE THREAD

**Issue by pkolmann** [simplesamlphp/simplesamlphp#1305]
We have a theme based on the new Symfony Twig framework. This also fails with the supplied standard theme when 'usenewui' is set to true.

Trying to parse an XML file to SSP metadata fails with an error.

**To Reproduce**
1. enable 'usenewui' in config.php
2. Go to 'Federation'
3. Click on 'XML to SimpleSAMLphp metadata converter'
4. Paste SP-XML-Metadata to text-area and click Parse
5. Error happens

**Expected behavior**
metadata php array should be shown

**Screenshots or logs**
Following error is shown:

```
SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Backtrace:
1 www/_include.php:17 (SimpleSAML_exception_handler)
0 [builtin] (N/A)
Caused by: InvalidArgumentException: An uploaded file must be an array or an instance of UploadedFile.
Backtrace:
6 vendor/symfony/http-foundation/FileBag.php:49 (Symfony\Component\HttpFoundation\FileBag::set)
5 vendor/symfony/http-foundation/FileBag.php:61 (Symfony\Component\HttpFoundation\FileBag::add)
4 vendor/symfony/http-foundation/FileBag.php:40 (Symfony\Component\HttpFoundation\FileBag::replace)
3 vendor/symfony/http-foundation/FileBag.php:31 (Symfony\Component\HttpFoundation\FileBag::__construct)
2 vendor/symfony/http-foundation/Request.php:261 (Symfony\Component\HttpFoundation\Request::initialize)
1 lib/SimpleSAML/Module.php:183 (SimpleSAML\Module::process)
0 www/module.php:10 (N/A)
```

**Comment by RosemaryOrchard** [comment]
It appears to me that the problem stems from this:

```
        if ($xmlfile = $request->files->get('xmlfile')) {
            $xmldata = trim(file_get_contents($xmlfile->getPathname()));
        } elseif ($xmldata = $request->request->get('xmldata')) {
            $xmldata = trim($xmldata);
        }
```

([Code here](https://github.com/simplesamlphp/simplesamlphp/blob/master/modules/admin/lib/Controller/Federation.php#L351)

Unfortunately flipping the logic to check for the form field before the file doesn't work.

**Comment by tvdijen** [comment]
Are you using 1.18? I'm unable to reproduce this..

**Comment by jaimeperez** [comment]
Hi @pkolmann!

I'm just trying my luck here, but could this be a dependency issue? What happens if you run `composer update` in your SimpleSAMLphp installation? Does the error still happen?

**Comment by pkolmann** [comment]
> Are you using 1.18? I'm unable to reproduce this..

Yes, just downloaded a fresh install of 1.18.5 and it's still the same with my config. I have my Config in an extra folder and redirect there with an ENV variable.

I've added my config as a JSON:
[config.json.txt](https://github.com/simplesamlphp/simplesamlphp/files/4409987/config.json.txt)

**Comment by pkolmann** [comment]
@tvdijen are you uploading a file or pasting the XML in the textbox?

It works for me as well, when uploading a file but crashes with the XML pasted into the textbox.

**Comment by pkolmann** [comment]
I've debugged it on my side now.

While taking a look at the current code, I found it was fixed already:

https://github.com/simplesamlphp/simplesamlphp/blob/master/lib/SimpleSAML/Module.php#L188

This has been fixed in 
* https://github.com/simplesamlphp/simplesamlphp/commit/f6e8c23f3e856c4b666bbe443a30ca6dabb3dadf
* https://github.com/simplesamlphp/simplesamlphp/commit/3ad5dfaf869566b6bbe4560544e6d40d9c3c9a87
* https://github.com/simplesamlphp/simplesamlphp/commit/300f74eef8bcda3d17e22d67d60e3df7453bafe2

for master branch.

**Comment by tvdijen** [comment]
Closed in #1306
